client.go

```diff
@@ -502,6 +503,9 @@ func (r *backoffReconnect) reconnect(c *Client) error {
 		reconnects++
 		err := c.doReconnect()
 		if err != nil {
+			if err == ErrClientDestroyed {
+				return err
```

Comment:
This looks like clean exit because client was closed and no reconnect required - maybe we should not return error here? Just nil